marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/survivorship/merging/base.xqy

```diff
@@ -2317,12 +2317,15 @@ declare function merge-impl:is-uri-locked($uri as xs:string) as xs:boolean {
   $uri = merge-impl:locked-uris()
 };
 
+(: Don't want to trigger static analysis for our sepatate read-only operations :)
+declare variable $lock-for-update-fun := fn:function-lookup(xs:QName('xdmp:lock-for-update'), 1);
```

Comment:
Interesting - so without this, there's a chance that the matching code may hit this function and thus ML will make it an update function?

I'm wondering if the entrance to the matching function should fail if xdmp:request-timestamp() doesn't return a timestamp so that it fails fast and there's no chance of it running in update mode?

(Also, tiny thing - "sepatate" should be "separate")

Reply:
Fixed typo and we call lock-for-update early on with our mastering step already.